cheroot/server.py

```diff
@@ -1204,6 +1254,190 @@ def send_headers(self):
         self.conn.wfile.write(EMPTY.join(buf))
 
 
+class HyperHTTPRequest(HTTPRequest):
+    """An HTTP Request (and response).
+
+    A single HTTP connection may consist of multiple request/response pairs.
+    """
+
+    server = None
+    """The HTTPServer object which is receiving this request."""
+
+    conn = None
+    """The HTTPConnection object on which this request connected."""
+
+    inheaders = {}
+    """A dict of request headers."""
```

Comment:
Should this be a dict, though? I seem to recall that there could be header duplicates...